packages/kbn-config-schema/src/internals/index.ts

```diff
@@ -274,6 +274,54 @@ export const internals = Joi.extend([
       },
     ],
   },
+  {
+    name: 'record',
```

Comment:
__note:__ it's quite similar to what we do for `map`, but I decided to not generalize here and let it live separately for a while.